include/circt/Dialect/SV/SVExpressions.td

```diff
@@ -91,3 +91,22 @@ def ConstantZOp : SVOp<"constantZ", [NoSideEffect, HasCustomSSAName]> {
   let results = (outs HWIntegerType:$result);
   let assemblyFormat = " attr-dict `:` type($result)";
 }
+
+def LocalparamOp : SVOp<"localparam",
+        [NoSideEffect, FirstAttrDerivedResultType, HasCustomSSAName]> {
+  let summary = "Declare a localparam";
+  let description = [{
+    The localparam operation produces a localparam of standard integer type
+    without a sign.
+    ```
+      %result = hw.localparam 42 : t1
+    ```
+    }];
+
+  let arguments = (ins APIntAttr:$value, StrAttr:$name);
+  let results = (outs HWIntegerType:$result);
```

Comment:
can sv local params bind to x or z?  If so we should allow that.
What about vectors, structures, and strings?  Those all can be localparam types.  Supporting those shouldn't block this patch, but will likely be needed eventually.

Reply:
I agree with darth here -- the initializer should be a value operand, allowing an arbitrary initializer expression, rather than a hard coded integer.  This would avoid having to hard code in knowledge of Z and X, and will be important when we support real parameters, so we can say `localparam x = otherparam+1` or whatever

Reply:
localparam x = 1
localparam y = x + 1
should work even if only considering integer types.

Reply:
Makes sense. I'm digging into `ExportVerilog` to make this work.